### ISSUE TITLE
force require administrator rights

### ISSUE THREAD

**Issue by silentnightwalk** [oleg-shilo/wixsharp#168]
Sorry for disturbing

I'm trying to install a topshelf service. It is a windows service which is done with topshelf. 

I tried to do the following:
```
new WixSharp.ServiceInstaller($@"MyService.exe", "install")
```
however this did not work out well because ServiceInstaller does not take two arguments.

The following worked out well, but only once:
```
new WixSharp.InstalledFileAction($@"MyService.exe", "install")
```

When i tried to repeat it again it did not work. First time i saw the request for administrator rights and I accepted it. Second time there was no request for administrator rights. Lack of administrator rights resulted in the fact that product was installed successfully, but the service was not installed (service is the part of the product).

Is it possible to force require administrator rights?

**Comment by silentnightwalk** [comment]
The following approach did not succeed. Installer did not install.

```
        [CustomAction]
        public static ActionResult InstallService(Session session)
        {
            Process proc = new Process();
            proc.StartInfo.FileName = $@"{TargetFolder}\{_serviceProductName}_{_productSufix}\MS4DService.exe";
            proc.StartInfo.Arguments = "install start";
            proc.StartInfo.Verb = "runas";
            proc.Start();
            return ActionResult.Success;
        }
```

**Comment by oleg-shilo** [comment]
Have a look at "WinService_WiX" sample. It shows how to use 'ServiceInstaller' class. In your sample you are not using it as intended.

As for `public static ActionResult InstallService(Session session)`. You need to be elevated to install any service. Thus your action needs to be deferred (MSI slang). `AfterInstall ` gives you an instant elevation:

```C#
project.AfterInstall += (SetupEventArgs e)=>
{
    ...
};
```

**Comment by silentnightwalk** [comment]
Big Thanks!
